lightning/src/onion_message/functional_tests.rs

```diff
@@ -115,6 +122,12 @@ impl CustomOnionMessageHandler for TestCustomMessageHandler {
 	}
 }
 
+impl ResponseErrorHandler for TestCustomMessageHandler {
```

Comment:
Why do we need this additional trait, instead of returning a `Result` from `handle_custom_message`? We could also add an event for failed OM handling.

Reply:
The error happens after calling `handle_custom_message` when `OnionMessenger` attempts to send the reply. This avoid reentrancy when if instead the user was given a reference to `OnionMessenger` to directly send the reply.

Reply:
Would it be reasonable to implement `EventsProvider` for `Messenger` and generate OM failure events there? Seems in line with what we do for outbound payment path failures

Reply:
Possibly... @TheBlueMatt is there any concern here? I'd imagine we may want to avoid onion messages from an untrusted source filling our event queue at no cost. At very least the cost for an outbound payment is to tie up liquidity.

Reply:
Discussed offline. For now we'll leave the trait but may want to have the `ChannelManager` implementation produce an event in the future.